client/storage/index.js

```diff
@@ -166,23 +226,22 @@ const uploadChunk = async data => {
         chunk.dl_status = DOWNLOAD_UPLOAD_STATUS.IN_PROGRESS;
         await chunk.save();
 
-        const formData = new FormData();
-        formData.append('file', data);
-        formData.append('__pn_integration_version_major', config.arweave_experiment_version_major);
-        formData.append('__pn_integration_version_minor', config.arweave_experiment_version_minor);
-        formData.append('__pn_chunk_id', chunkId);
-        formData.append(
-            `__pn_chunk_${config.arweave_experiment_version_major}.${config.arweave_experiment_version_minor}_id`,
-            chunkId
-        );
-
-        const response = await axios.post(
-            `${config.arweave_bundler_url}/signPOST`,
-            formData,
-            {headers: {...formData.getHeaders()}}
-        );
-
-        // TODO: check status from bundler
+        const chunkIdVersioned = `__pn_chunk_${config.arweave_experiment_version_major}.${config.arweave_experiment_version_minor}_id`;
+
+        const tags = {
+            __pn_integration_version_major: config.arweave_experiment_version_major,
+            __pn_integration_version_minor: config.arweave_experiment_version_minor,
+            __pn_chunk_id: chunkId,
+            [chunkIdVersioned]: chunkId
+        };
+
+        let response;
+        if (process.env.MODE === 'zappdev')
+            response = await uploadArweave(data, tags);
+        else
+            response = await uploadBundler(data, tags);
+        
```

Comment:
@diogosmendonca ok let's discuss this together.

Reply:
@diogosmendonca generally you're right, but in this particular case we are mixing development tools and a real-life workflow together in one file, which brings confusion and unnecessary complexity. It is ok to have two similar module copies in this case, since only one of them will only be used in production while the other one is for testing. This way it will be way much easier to figure out its' purpose.

Reply:
The ideal solution is to have only one codebase. We do development/testing in the same codebase (or as near as possible) that will be put in production to have quality grantee that everything will work fine when goes live. The ZApp developers should experience that kind of seamless transition to development to production of their ZApps. When we split code of ZApp development environment and production one, we are given chance of ZApps that works in development environment fails when go to production and we would have maintenance problems of the codebase, as I stated before. 

I summarize bellow the current differences between production and development environments. They are all in storage module and are guarded by two config variables, storage.use_arlocal and storage.use_arweave_bundler, which the values are set by default for the production setup, i.e., storage.use_arlocal=false and storage.use_arweave_bundler=true. The changes are in this points:

[Init](https://github.com/pointnetwork/pointnetwork/blob/926e89ac0ef5d4a9a7c6c4179f8b8e1e4941c730/client/storage/index.js#L80) - to mint tokens for arlocal
[uploadChunk](https://github.com/pointnetwork/pointnetwork/blob/926e89ac0ef5d4a9a7c6c4179f8b8e1e4941c730/client/storage/index.js#L124) - to upload chunks using or not the bundler.
[getChunk](https://github.com/pointnetwork/pointnetwork/blob/926e89ac0ef5d4a9a7c6c4179f8b8e1e4941c730/client/storage/index.js#L243) - to bypass an issue of arlocal when downloading files (this will be removed as soon as the bug of arlocal is fixed) 

To change this code to other file we will need to duplicate at least the methods mentioned above. However, as getChunk and uploadChunk are not exported methods, we also need to duplicate the code of exported method that uses them, in case, getFile and uploadFlie. As uploadDir is also exported and uses uploadFile it will need to be copied also. The same process occurs with getJSON and getFileIdByPath. This will produce the whole storage/index.js duplicated, which has approximately 600 lines of code. This don't seems to be a good design decision.  

@jensendarren and @tankakatan let me know about the design decision for topic so that I could implement and finish this PR.

Reply:
@diogosmendonca I've gone ahead and created a separate `index-arlocal.js` file for now. This is because, the `zappdev` environment is mainly for Zapp developers to build and test the Zapp functionality in a fast and scaled down environment. The general procedure should be that when the Zapp developer is ready to deploy to Production they would first test their Zapp in the new `e2e` environment which closely matches Production and it will be here they would identitfy any issues when using the 'real' Arweave and the full end to end deployment process. Moreover, the `storage/index.js` file is now so different again that its best to keep these two files completely separate now. As long as `arlocal` works we should be fine and the main storage index file can continue to evolve separately from the `arlocal` version.